lib/rules/media-feature-name-no-unknown/index.js

```diff
@@ -45,46 +45,45 @@ const rule = (primary, secondaryOptions) => {
 		}
 
 		root.walkAtRules(/^media$/i, (atRule) => {
-			mediaParser(atRule.params).walk(/^media-feature$/i, (mediaFeatureNode) => {
-				const parent = mediaFeatureNode.parent;
-				const mediaFeatureRangeContext = isRangeContextMediaFeature(parent.value);
-
-				let value;
-				let sourceIndex;
-
-				if (mediaFeatureRangeContext) {
-					const parsedRangeContext = rangeContextNodeParser(mediaFeatureNode);
-
-					value = parsedRangeContext.name.value;
-					sourceIndex = parsedRangeContext.name.sourceIndex;
-				} else {
-					value = mediaFeatureNode.value;
-					sourceIndex = mediaFeatureNode.sourceIndex;
-				}
-
-				if (!isStandardSyntaxMediaFeatureName(value) || isCustomMediaQuery(value)) {
-					return;
-				}
-
-				if (optionsMatches(secondaryOptions, 'ignoreMediaFeatureNames', value)) {
-					return;
-				}
-
-				if (vendor.prefix(value) || mediaFeatureNames.has(value.toLowerCase())) {
-					return;
-				}
-
-				const index = atRuleParamIndex(atRule) + sourceIndex;
-				const endIndex = index + value.length;
-
-				report({
-					index,
-					endIndex,
-					message: messages.rejected,
-					messageArgs: [value],
-					node: atRule,
-					ruleName,
-					result,
+			const mediaQueryList = parseMediaQuery(atRule);
+
+			mediaQueryList.forEach((mediaQuery) => {
+				if (isMediaQueryInvalid(mediaQuery)) return;
+
+				mediaQuery.walk((entry) => {
+					const node = entry.node;
+
```

Comment:
[trivial] Shortening code a bit may also be good, but keeping is OK.

```suggestion
				mediaQuery.walk(({ node }) => {
```